src/main/java/com/bookshop/controller/BookController.java

```diff
@@ -29,7 +36,19 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    @ResponseStatus(HttpStatus.CREATED)
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
+
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public BookDto updateBook(@PathVariable Long id,
+                              @RequestBody @Valid CreateBookRequestDto requestDto) {
+        return bookService.update(id, requestDto);
+    }

```

Comment:
Please, add validation to public BookDto getBookById(@PathVariable @Positive Long id) {

Reply:
I have this in the next branch, it will be added when I merge